src/lib/datepicker/datepicker.ts

```diff
@@ -116,6 +117,17 @@ export class MdDatepicker<D> implements OnDestroy {
    */
   @Input() touchUi = false;
 
+  /** Whether the datepicker pop-up should be disabled. */
+  @Input()
+  get disabled() { return this._disabled; }
+  set disabled(value: any) {
+    this._disabled = coerceBooleanProperty(value);
+    if (this._disabled === undefined) {
```

Comment:
similar to how i showed in the other example, put the delegation logic in the getter